### ISSUE TITLE
Remote Ray Dataloader

### ISSUE THREAD

**Issue by Jay-ju** [ray-project/ray#51393]
### Description
Related to this issue at https://github.com/ray-project/ray/issues/51392. The remote dataloader is an open interface of Pytorch Dataloader. Here, it is temporarily named TorchBasedRayDataloader. The implementation principle of the demo is as follows. In this way, the allocation logic of the dataloader can be not intruded. Ray is only regarded as a worker of torch. In addition to this advantage, it can also solve the problem that torch can only be limited by the resources of the current node. However, the following example is only for demonstration. Among them, problems such as order, GIL lock, and resume need to be handled. I wonder if the community is interested in implementing a remote version of torch dataloader. If interested, I can contribute here.
```
@ray.remote(num_cpus=1)
class RemoteIterDatasetActor(IterableDataset):
    """Actually get data from remote loader server 

    extend : get plan data from Server.
    """ 
        
    def __init__(
            self,
            torch_dataloader: DataLoader
    ):  
        self.torch_dataloader = torch_dataloader
           
    def __iter__(self):
        return iter(self.torch_dataloader)      

class DataloaderBasedRay():
    
    def __init__(self, torch_dataloader) -> None:
        super().__init__()
        self.dataset = RemoteRayIterDataset(torch_dataloader.dataset)
        self.queue = queue.Queue()
        ray.init()
        remote_dataset_actor = RemoteIterDatasetActor \
            .options(num_cpus=DATA_LOAD_ACTOR_NUM_CPUS, scheduling_strategy="SPREAD") \
            .remote(torch_dataloader)
        prefetch_thread = threading.Thread(
            target=self.__prefetch, daemon=True)
        prefetch_thread.start() 
        
        
    def __prefetch():
        result_ref = remote_dataset_actor.__iter__.remote()
        self.queue.put(ray.get(result_ref))
            
        
    def __iter__(self): 
        self.queue.pop()
```

_No response_

**Comment by richardliaw** [comment]
Closing this out for now, since there's no action required on ray side
